test/karma/dataBehaviorSpec.js

```diff
@@ -0,0 +1,1681 @@
+          });
+          initialContextModel.trigger('change:id');
+        });
+    });
+
+    it('and depends on another behavior will re-fetch data when the ids on the root behavior change:\n\
+ids = {\n\
+  property: \'behaviors.dataBehavior.otherIds\'\n\
+}\n\\n\
+', function(done) {
+      $.mockjax.clear(this.routes['/myModel/ids|post']);
+      $.mockjax(MOCKJAX_ROUTE_WITH_OTHER_IDS);
+
+      var defaultBehaviorConfiguration = getBasicBehaviorConfiguration();
+      defaultBehaviorConfiguration.returnSingleResult = true;
+      defaultBehaviorConfiguration.ids = {
+        property: 'viewState.idFromView'
+      };
+      var defaultBehavior2Configuration = getBasicBehaviorConfiguration();
+      defaultBehavior2Configuration.ids = {
+        property: 'behaviors.dataBehavior.otherIds'
+      };
+      var ViewWithBehavior = TorsoView.extend({
+        behaviors: {
+          dataBehavior: defaultBehaviorConfiguration,
+          dataBehavior2: defaultBehavior2Configuration
+        }
+      });
+      var viewWithBehavior = new ViewWithBehavior();
+      var dataBehavior = viewWithBehavior.getBehavior('dataBehavior');
+      dataBehavior.once('fetched', function() {
+        expect(dataBehavior.toJSON()).toEqual({ id: 10, count: 0, otherIds: [1, 2, 3], otherOtherIds: ['a0', 'b0', 'c0', 'd0', 'e0', 'f0'] });
+      });
+
+      var dataBehavior2 = viewWithBehavior.getBehavior('dataBehavior2');
+      dataBehavior2.once('fetched', function() {
+        expect(dataBehavior.toJSON()).toEqual({ id: 10, count: 0, otherIds: [1, 2, 3], otherOtherIds: ['a0', 'b0', 'c0', 'd0', 'e0', 'f0'] });
+        expect(dataBehavior2.toJSON()).toEqual([{ id: 1, count: 0, otherIds: [1, 2, 3], otherOtherIds: ['a0', 'b0', 'c0', 'd0', 'e0', 'f0'] },
+                                                { id: 2, count: 1, otherIds: [2, 3, 4], otherOtherIds: ['a1', 'b1', 'c1', 'd1', 'e1', 'f1'] },
+                                                { id: 3, count: 2, otherIds: [3, 4, 5], otherOtherIds: ['a2', 'b2', 'c2', 'd2', 'e2', 'f2'] }]);
+        done();
+      });
+      viewWithBehavior.set('idFromView', 10);
+    });
+
+    it('will handle a collection of models each with an array of ids:\n\
+ids = {\n\
+  property: \'behaviors.dataBehavior.otherIds\'\n\
+}\n\
+\n\
+dataBehavior gets multiple objects with this structure: { id: 10, otherIds: [1, 2, 3] ... }.\n\
+dataBehavior2 will pull the unique collection of ids collected from all of the otherIds in dataBehavior\'s objects.\n\
+\n\
+', function(done) {
+      $.mockjax.clear(this.routes['/myModel/ids|post']);
+      $.mockjax(MOCKJAX_ROUTE_WITH_OTHER_IDS);
+
+      var defaultBehaviorConfiguration = getBasicBehaviorConfiguration();
+      defaultBehaviorConfiguration.ids = {
+        property: 'viewState.idFromView'
+      };
+      var defaultBehavior2Configuration = getBasicBehaviorConfiguration();
+      defaultBehavior2Configuration.ids = {
+        property: 'behaviors.dataBehavior.otherIds'
```

Comment:
yea, but what's the [something]?

I updated the data behavior to proxy all events from the private collection (which includes change events).  I think this is cleaner and is along the same lines with how toJSON() was updated.